DataFormats/TrackerRecHit2D/src/ProjectedSiStripRecHit2D.cc

```diff
@@ -5,6 +5,6 @@
 
 void ProjectedSiStripRecHit2D::setDet(const GeomDet& idet) {
   TrackingRecHit::setDet(idet);
-  const GluedGeomDet& gdet = static_cast<const GluedGeomDet&>(idet);
+  const GluedGeomDet& gdet = dynamic_cast<const GluedGeomDet&>(idet);
```

Comment:
it cannot be nothing else. It is correct by construction.
This is production code, not a debugging tool